### ISSUE TITLE
Node js error at startup: 'Cannot read property 'close' of undefined'

### ISSUE THREAD

**Issue by Sholofly** [danielwelch/hassio-zigbee2mqtt#97]
Hi Daniel,
Some people, including me,  are experiencing the issue below. I've tried te restart and resinstall the addon on hassio but without luck. Can you take a look at this?

p.s.: The edge version has the same error.

HA: 0.86.3
add-on: 1.1.0

@danielwelch 
I'm getting this error now:
``` 
zigbee2mqtt:info 2019-1-31 06:14:03 Error while starting zigbee-shepherd, attemping to fix... (takes 60 seconds)
/zigbee2mqtt-1.1.0/node_modules/q/q.js:155
                throw e;
                ^
TypeError: Cannot read property 'close' of undefined
    at shepherd.start (/zigbee2mqtt-1.1.0/lib/zigbee.js:45:47)
    at /zigbee2mqtt-1.1.0/node_modules/q/q.js:2059:17
    at runSingle (/zigbee2mqtt-1.1.0/node_modules/q/q.js:137:13)
    at flush (/zigbee2mqtt-1.1.0/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)```

_Originally posted by @noxhirsch in https://github.com/danielwelch/hassio-zigbee2mqtt/issues/95#issuecomment-459218637_

**Comment by danielwelch** [comment]
Thanks for opening up a new issue and providing more info. Would you mind posting your add-on configuration as well?

**Comment by Claypaenguin** [comment]
I'm having the same issue although it didn't happen right off the bat after updating, but after a couple of hours.

If you want to have a look at my config (I deleted my devices)
```
advanced:
  channel: 25
  log_level: debug
  network_key:
  - 17
  - 12
  - 76
  - 207
  - 51
  - 2
  - 79
  - 13
  - 195
  - 16
  - 243
  - 43
  - 5
  - 133
  - 69
  - 17
devices:
homeassistant: true
mqtt:
  base_topic: zigbee2mqtt
  password: password
  server: mqtt://homeassistant
  user: mqtt
permit_join: false
serial:
  port: /dev/ttyACM0
```

**Comment by danielwelch** [comment]
Thanks, if you could also post your JSON formatted config from the Hassio front end that would be great.

**Comment by Claypaenguin** [comment]
Sure thing.
```
{
  "data_path": "/share/zigbee2mqtt",
  "homeassistant": true,
  "permit_join": false,
  "mqtt_base_topic": "zigbee2mqtt",
  "mqtt_server": "mqtt://homeassistant",
  "serial_port": "/dev/ttyACM0",
  "devices": [],
  "network_key": []
}
```
Another thing I just remembered is that this error happened immediately after rapidly pressing a connected Hue dimmer. Not sure if that could've any impact.

**Comment by danielwelch** [comment]
Can you try with version `v1.1.0.1`? It's just been pushed to Docker hub

**Comment by mgsb** [comment]
Is there a way to make this happen via hass.io UI? Or need to ssh in and run some docker commands?

**Comment by danielwelch** [comment]
I just mean updating via the normal UI process

**Comment by mgsb** [comment]
Patiently waiting for it to show up. Thanks for the quick patch.

**Comment by Sholofly** [comment]
Just updated, but same result.

**Comment by raystream** [comment]
same failure here

**Comment by danielwelch** [comment]
May have to turn to @Koenkk for some help here. Seems like an error starting zigbee-shepherd, and then it throws shen attempting to handle that error here https://github.com/Koenkk/zigbee2mqtt/blob/master/lib/zigbee.js#L45

**Comment by Sholofly** [comment]
They have already addressed the problem and fixed it in the dev-branche:
[https://github.com/Koenkk/zigbee2mqtt/issues/971](https://github.com/Koenkk/zigbee2mqtt/issues/971)

**Comment by Sholofly** [comment]
Tested the work around without luck 🤕 ... @danielwelch Maybe you can create a edge release from the dev-branche of @Koenkk  I don't know if that's possible?

**Comment by danielwelch** [comment]
Ah, well the edge version of the add-on uses the dev branch so if that’s the case, should be good to go with the edge version

**Comment by Sholofly** [comment]
Oke, does the Instagram checkout the latest git branche? Cool!

**Comment by Koenkk** [comment]
Indeed not an hassio addon issue & should be fixed in `-edge`.

**Comment by Sholofly** [comment]
I've installed the edge-version. removed and reconnected the device. A different error. Because i can't figure out if it's a add-on related issue or an issue in the original code I place it here:

```
> zigbee2mqtt@1.1.0 start /app
> node index.js
		Zigbee2mqtt requires node version 10, you are running v8.14.0!
  zigbee2mqtt:info 2019-2-1 19:39:47 Logging to directory: '/share/zigbee2mqtt/log/2019-02-01.19-39-47'
  zigbee2mqtt:info 2019-2-1 19:39:48 Starting zigbee2mqtt version 1.1.0 (commit #dd8ee60)
  zigbee2mqtt:info 2019-2-1 19:39:48 Starting zigbee-shepherd
  zigbee2mqtt:info 2019-2-1 19:39:51 Error while starting zigbee-shepherd, attemping to fix... (takes 60 seconds)
  zigbee2mqtt:info 2019-2-1 19:40:51 Starting zigbee-shepherd
  zigbee2mqtt:error 2019-2-1 19:40:54 Error while starting zigbee-shepherd!
  zigbee2mqtt:error 2019-2-1 19:40:54 Press the reset button on the stick (the one closest to the USB) and start again
  zigbee2mqtt:error 2019-2-1 19:40:54 Failed to start
	{"message":"request timeout","stack":"Error: request timeout\n    at CcZnp.<anonymous> (/app/node_modules/cc-znp/lib/ccznp.js:255:22)\n    at Object.onceWrapper (events.js:315:30)\n    at emitOne (events.js:116:13)\n    at CcZnp.emit (events.js:211:7)\n    at Timeout.<anonymous> (/app/node_modules/cc-znp/lib/ccznp.js:234:18)\n    at ontimeout (timers.js:498:11)\n    at tryOnTimeout (timers.js:323:5)\n    at Timer.listOnTimeout (timers.js:290:5)"}
```

**Comment by Koenkk** [comment]
- stop zigbee2mqtt
- re-plug stick
- press reset button on the stick (the one close to the usb port), green light should go off
- start zigbee2mqtt

**Comment by Sholofly** [comment]
> 

> * stop zigbee2mqtt
> * re-plug stick
> * press reset button on the stick (the one close to the usb port), green light should go off
> * start zigbee2mqtt

Did that, now again, just to be sure. Gives me the same result.

**Comment by stratplayer** [comment]
I upgraded via the Hass.io UI and now it wont start and my config is reset to default :-(

`> zigbee2mqtt@1.1.0 start /zigbee2mqtt-1.1.0
> node index.js
  zigbee2mqtt:info 2019-2-1 21:25:15 Logging to directory: '/share/zigbee2mqtt/log/2019-02-01.21-25-15'
  zigbee2mqtt:info 2019-2-1 21:25:16 Starting zigbee2mqtt version 1.1.0 (commit #unknown)
  zigbee2mqtt:info 2019-2-1 21:25:16 Starting zigbee-shepherd
  zigbee2mqtt:info 2019-2-1 21:25:16 Error while starting zigbee-shepherd, attemping to fix... (takes 60 seconds)
/zigbee2mqtt-1.1.0/node_modules/q/q.js:155
                throw e;
                ^
TypeError: Cannot read property 'close' of undefined
    at shepherd.start (/zigbee2mqtt-1.1.0/lib/zigbee.js:45:47)
    at /zigbee2mqtt-1.1.0/node_modules/q/q.js:2059:17
    at runSingle (/zigbee2mqtt-1.1.0/node_modules/q/q.js:137:13)
    at flush (/zigbee2mqtt-1.1.0/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)
npm
 
ERR!`

**Comment by stratplayer** [comment]
I restored to a snapshot that was working

**Comment by Sholofly** [comment]
The edge version finally works for me. After the fix for the 'close' issue, the combination with my  zigbee usb stick wasn't successful. I managed to fix that.

**Comment by SpuriousGer** [comment]
I am experiencing the same issue. My zigbee devices are no longer changing their state (they are detected). I don't want to switch to the edge version as it is more work. What can I do? Do I need to do anything after updating to 1.1.1?

**Comment by DeliriousMetro** [comment]
Just had setup this up yesterday and was enjoying home automation until the Android HomeAssistant app glitched out on the brightness of a Phillips Hue bulb and now after restarting zigbee2mqtt, I get the close error.

Log error:
`  zigbee2mqtt:info 2019-2-3 17:27:03 Error while starting zigbee-shepherd, attemping to fix... (takes 60 seconds)
/zigbee2mqtt-1.1.0/node_modules/q/q.js:155
                throw e;
                ^
TypeError: Cannot read property 'close' of undefined
    at shepherd.start (/zigbee2mqtt-1.1.0/lib/zigbee.js:45:47)
    at /zigbee2mqtt-1.1.0/node_modules/q/q.js:2059:17
    at runSingle (/zigbee2mqtt-1.1.0/node_modules/q/q.js:137:13)
    at flush (/zigbee2mqtt-1.1.0/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)
npm
 ERR! 
code ELIFECYCLE
npm ERR!
 errno 1
npm
 ERR! zigbee2mqtt@1.1.0 start: `node index.js`
npm
 ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.1.0 start script.
npm 
ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR!
 A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-02-03T17_27_03_316Z-debug.log
2019-02-03T17:27:04: PM2 log: App [npm:0] exited with code [1] via signal [SIGINT]
2019-02-03T17:27:04: PM2 log: App [npm:0] starting in -fork mode-
2019-02-03T17:27:04: PM2 log: App [npm:0] online`

I currently do not have access to an Phillips dimmer, so I would be unable to reset these paired bulbs.

Just want to know if pressing the reset button on the CC2531 will loose my paired devices? Will updating to the addon loose my paired devices.

**Comment by Koenkk** [comment]
- Pressing the reset button on the CC2531 (the one closest to USB port) won't loose your paired devices
- Updating zigbee2mqtt won't loose your paired devices

Only updating (flashing) the firmware of the CC2531 looses your paired devices.

**Comment by DeliriousMetro** [comment]
> 
> 
>     * Pressing the reset button on the CC2531 (the one closest to USB port) won't loose your paired devices
> 
>     * Updating zigbee2mqtt won't loose your paired devices
> 
>     * Only updating (flashing) the firmware of the CC2531 looses your paired devices.

Thanks dude, I can confirm that has fixed the issue and the bulbs and Xiaomi sensors are all paired sucessfully on the normal zigbee2mqtt 1.1.0.1 hass.io addon.

I will try the edge version as well to confirm that no glitches that cause the close state are present.

**Comment by Koenkk** [comment]
@DeliriousMetro 1.1.1 should already be out for hassio.

**Comment by DeliriousMetro** [comment]
> 
> 
> @DeliriousMetro 1.1.1 should already be out for hassio.

Still on the old one for me, so ended up using the edge one.

![image](https://user-images.githubusercontent.com/9902130/52180743-1a5e7180-27e2-11e9-89bc-138c72cdeab6.png)

I'll try to reinstall it now, maybe that would trigger an update check.

**Comment by Koenkk** [comment]
@DeliriousMetro sorry, should be up in an hour.

**Comment by DeliriousMetro** [comment]
> @DeliriousMetro sorry, should be up in an hour. (will be up when https://travis-ci.org/danielwelch/hassio-zigbee2mqtt/builds/488214025?utm_source=github_status&utm_medium=notification is done)

Do not worry, I appreciate the time and work you have put into this. Especially, as a neat standalone hass.io addon. Thanks for the quick fix too, for me, resetting the usb did work on both builds so I will be fine to update whenever the build has been passed.

**Comment by SpuriousGer** [comment]
It worked for me again, but I have the problem now that one sensor which is a bit far away won't connect anymore. It was working previously flawlessly, but now I can't get it to send signals. It's a door sensor and I've already mounted it to the door. Anyone with an idea on how to get it to be recognized again?

**Comment by danielwelch** [comment]
I'll go ahead and close this issue since it seems things have been resolved with zigbee2mqtt version 1.1.1
